File: problem056.scala
import scala.math._
val limit = 100;
val ls = for (a <- 1 until limit; b <- 1 until limit) yield (a, b);
val l1 = ls map(x => pow(x._1.toDouble, x._2.toDouble).toString.toList.map(_.toInt - 48).takeWhile(_>0).sum);
println(l1.max);

File: problem056.py
#!/usr/bin/env python

limit = 100
max = 0
for a in xrange (1,limit):
    for b in xrange (1, limit):
        n = str(a**b)
        sum = 0
        for i in n:
            if i != 'L':
                sum += int(i)
        if sum > max: max = sum
#        print a, b, n, sum, max
print max
